### PR TITLE
Add MSI Support for Azure Container Instance

### PR DESCRIPTION
**Pull request by allan-xiazhan**
## Description

Add MSI Support for Azure Container Instance

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

**Comment by cormacpayne** [comment]
Design review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/208

File: src/ResourceManager/ContainerInstance/Commands.ContainerInstance/Commands/ContainerInstanceCmdletBase.cs [hunk 1]
```diff
@@ -140,15 +140,17 @@ public ContainerGroup CreateContainerGroup(ContainerGroupCreationParameters para
                 Tags = parameters.Tags,
                 Containers = new List<Container>() { container },
                 OsType = parameters.OsType,
-                RestartPolicy = parameters.RestartPolicy
+                RestartPolicy = parameters.RestartPolicy,
+                Identity = parameters.Identity
             };
 
-            if (string.Equals(IpAddress.Type, parameters.IpAddressType, StringComparison.OrdinalIgnoreCase) || 
+            if (string.Equals("Public", parameters.IpAddressType, StringComparison.OrdinalIgnoreCase) ||
```
[the remark this answers is not in the document]
Comment (allan-xiazhan) [reply src/ResourceManager/ContainerInstance/Commands.ContainerInstance/Commands/ContainerInstanceCmdletBase.cs]: The 'Type' property of IpAddress doesn't exist any more in the new SDK.  The value of that property was "Public".

**Comment by cormacpayne** [comment]
@allan-xiazhan I re-targeted your PR to our new default branch, `master`, so you will need to pull from that branch in order to resolve the merge conflicts.

_Note_: since the next release is the GA of the `Az` module, we have removed all reference docs to `AzureRM` cmdlets, so you will need to regenerate your help based on the `Az` cmdlets.

It appears the build error with StaticAnalysis is a false-positive, so you can follow [the following document](https://github.com/Azure/azure-powershell/blob/master/documentation/Debugging-StaticAnalysis-Errors.md#breaking-changes) on how to suppress this exception.

File: src/ResourceManager/ContainerInstance/Commands.ContainerInstance.Test/ScenarioTests/ContainerInstanceTests.cs [hunk 2]
```diff
@@ -60,7 +74,31 @@ public void TestContainerInstanceLog()
         [Trait(Category.AcceptanceType, Category.CheckIn)]
         public void TestCreateContainerGroupWithVolume()
         {
-            TestController.NewInstance.RunPowerShellTest(_logger, "Test-AzureRmContainerGroupWithVolumeMount");
+            TestController.NewInstance.RunPowerShellTest(_logger, "Test-AzureRmContainerGroupWithVolume");
+        }
+
+#if NETSTANDARD
```
Comment (cormacpayne) [inline src/ResourceManager/ContainerInstance/Commands.ContainerInstance.Test/ScenarioTests/ContainerInstanceTests.cs]: @allan-xiazhan we can remove these ifdefs, right? The `-Command` parameter is enabled in .NET Standard as of this PR

Comment (allan-xiazhan) [reply src/ResourceManager/ContainerInstance/Commands.ContainerInstance.Test/ScenarioTests/ContainerInstanceTests.cs]: I believe so. Good catch!

**Pull request merged**